diff -purN linux-4.17.orig/arch/x86/kernel/reboot.c linux-4.17/arch/x86/kernel/reboot.c
--- linux-4.17.orig/arch/x86/kernel/reboot.c	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/arch/x86/kernel/reboot.c	2018-06-08 15:06:13.367346310 +0200
@@ -654,6 +654,15 @@ static void native_machine_emergency_res
 void native_machine_shutdown(void)
 {
 	/* Stop the cpus and apics */
+
+#ifdef CONFIG_SMP
+	/*
+	 * Disable the local irq to not receive the per-cpu timer interrupt
+	 * which may trigger scheduler's load balance.
+	 */
+	local_irq_disable();
+#endif
+
 #ifdef CONFIG_X86_IO_APIC
 	/*
 	 * Disabling IO APIC before local APIC is a workaround for
@@ -671,11 +680,8 @@ void native_machine_shutdown(void)
 
 #ifdef CONFIG_SMP
 	/*
-	 * Stop all of the others. Also disable the local irq to
-	 * not receive the per-cpu timer interrupt which may trigger
-	 * scheduler's load balance.
+	 * Stop all of the others.
 	 */
-	local_irq_disable();
 	stop_other_cpus();
 #endif
 
