diff -purN linux-4.17.orig/include/linux/mfd/axp20x.h linux-4.17/include/linux/mfd/axp20x.h
--- linux-4.17.orig/include/linux/mfd/axp20x.h	2018-06-03 23:15:21.000000000 +0200
+++ linux-4.17/include/linux/mfd/axp20x.h	2018-06-08 15:06:55.515222803 +0200
@@ -181,8 +181,8 @@ enum axp20x_variants {
 #define AXP20X_BATT_CHRG_I_L		0x7b
 #define AXP20X_BATT_DISCHRG_I_H		0x7c
 #define AXP20X_BATT_DISCHRG_I_L		0x7d
-#define AXP20X_IPSOUT_V_HIGH_H		0x7e
-#define AXP20X_IPSOUT_V_HIGH_L		0x7f
+#define AXP20X_APS_V_H			0x7e
+#define AXP20X_APS_V_L			0x7f
 
 /* Power supply */
 #define AXP20X_DCDC_MODE		0x80
