diff -purN pppd-2.4.7.orig/pppd/plugins/rp-pppoe/plugin.c pppd-2.4.7/pppd/plugins/rp-pppoe/plugin.c
--- pppd-2.4.7.orig/pppd/plugins/pppoe/plugin.c	2018-01-30 23:14:51.744756634 +0100
+++ pppd-2.4.7/pppd/plugins/pppoe/plugin.c	2018-01-30 23:22:05.263938309 +0100
@@ -465,15 +465,6 @@
     conn->storedmtu = lcp_allowoptions[0].mru;
     conn->storedmru = lcp_wantoptions[0].mru;
 
-    ccp_allowoptions[0].deflate = 0;
-    ccp_wantoptions[0].deflate = 0;
-
-    ipcp_allowoptions[0].neg_vj = 0;
-    ipcp_wantoptions[0].neg_vj = 0;
-
-    ccp_allowoptions[0].bsd_compress = 0;
-    ccp_wantoptions[0].bsd_compress = 0;
-
     conn->discoveryTimeout = pppoe_padi_timeout;
     conn->discoveryAttempts = pppoe_padi_attempts;
 }
