--- python-mwscrape2slob-e01d3e92f0a372ebd0f57390e437a28f9d3c0438/mwscrape2slob/__init__.py.orig	2020-02-18 04:26:20.000000000 +0100
+++ python-mwscrape2slob-e01d3e92f0a372ebd0f57390e437a28f9d3c0438/mwscrape2slob/__init__.py	2022-12-30 20:37:43.137174527 +0100
@@ -283,7 +283,7 @@
                 for item in articles_from_viewiter(viewiter):
                     yield item
 
-        pool = multiprocessing.Pool(None, process_initializer, [
+        pool = multiprocessing.Pool(6, process_initializer, [
             self.filters,
             self.interwikimap,
             self.namespaces
