diff -uNr igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2.orig/lib/igt_halffloat.c igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2/lib/igt_halffloat.c
--- igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2.orig/lib/igt_halffloat.c	2021-04-23 12:38:04.000000000 +0200
+++ igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2/lib/igt_halffloat.c	2021-06-08 22:15:36.984081953 +0200
@@ -162,7 +162,7 @@
 	return fi.f;
 }
 
-#if defined(__x86_64__) && !defined(__clang__)
+#if defined(__x86_64__) && !defined(__clang__) && !defined(__UCLIBC__)
 #pragma GCC push_options
 #pragma GCC target("f16c")
 
diff -uNr igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2.orig/lib/igt_x86.c igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2/lib/igt_x86.c
--- igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2.orig/lib/igt_x86.c	2021-04-23 12:38:04.000000000 +0200
+++ igt-gpu-tools-igt-gpu-tools-1.26-203def046b466fb2da67f9f15552d84e1c0b41f2/lib/igt_x86.c	2021-06-08 22:15:06.071848819 +0200
@@ -190,7 +190,7 @@
 }
 #endif
 
-#if defined(__x86_64__) && !defined(__clang__)
+#if defined(__x86_64__) && !defined(__clang__) && !defined(__UCLIBC__)
 #pragma GCC push_options
 #pragma GCC target("sse4.1")
 #pragma GCC diagnostic ignored "-Wpointer-arith"
