Upstream: https://github.com/signalwire/freeswitch/issues/2202#event-6999616531

diff --git a/src/mod/applications/mod_av/avformat.c b/src/mod/applications/mod_av/avformat.c
index d617a52..edba9c2 100644
--- a/src/mod/applications/mod_av/avformat.c
+++ b/src/mod/applications/mod_av/avformat.c
@@ -637,12 +637,14 @@ static switch_status_t add_stream(av_file_context_t *context, MediaStream *mst,
 		c->rc_initial_buffer_occupancy = buffer_bytes * 8;
 
 		if (codec_id == AV_CODEC_ID_H264) {
+#if (LIBAVCODEC_VERSION_MAJOR < LIBAVFORMAT_8_V)
 #if ((LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_6_V && LIBAVFORMAT_VERSION_MINOR >= LIBAVFORMAT_61_V) || LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_7_V)
 GCC_DIAG_OFF(deprecated-declarations)
 #endif
 			c->ticks_per_frame = 2;
 #if ((LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_6_V && LIBAVFORMAT_VERSION_MINOR >= LIBAVFORMAT_61_V) || LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_7_V)
 GCC_DIAG_ON(deprecated-declarations)
+#endif
 #endif
 
 			c->flags|=AV_CODEC_FLAG_LOOP_FILTER;   // flags=+loop
@@ -1431,7 +1433,7 @@ static switch_status_t open_input_file(av_file_context_t *context, switch_file_h
 		switch_goto_status(SWITCH_STATUS_FALSE, err);
 	}
 
-#if (LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_7_V)
+#if (LIBAVFORMAT_VERSION_MAJOR >= LIBAVFORMAT_7_V)
 	handle->seekable = !(context->fc->iformat->flags & AVFMT_NOTIMESTAMPS);
 #else
 	handle->seekable = context->fc->iformat->read_seek2 ? 1 : (context->fc->iformat->read_seek ? 1 : 0);
@@ -3123,14 +3125,18 @@ static switch_status_t av_file_read_video(switch_file_handle_t *handle, switch_f
 	void *pop;
 	MediaStream *mst = &context->video_st;
 	AVStream *st = mst->st;
+#if (LIBAVCODEC_VERSION_MAJOR < LIBAVFORMAT_8_V)
 	int ticks = 0;
+#endif
 	int64_t max_delta = 1 * AV_TIME_BASE; // 1 second
 	switch_status_t status = SWITCH_STATUS_SUCCESS;
 	double fl_to = 0.02;
 	int do_fl = 0;
 	int smaller_ts = context->read_fps;
+#if (LIBAVCODEC_VERSION_MAJOR < LIBAVFORMAT_8_V)
 	AVCodecContext *c = NULL;
 	AVCodecParserContext *cp = NULL;
+#endif
 
 	if (!context->has_video) return SWITCH_STATUS_FALSE;
 
@@ -3238,6 +3244,7 @@ static switch_status_t av_file_read_video(switch_file_handle_t *handle, switch_f
 	}
 #endif
 
+#if (LIBAVCODEC_VERSION_MAJOR < LIBAVFORMAT_8_V)
 	if ((c = av_get_codec_context(mst)) && c->time_base.num) {
 		cp = av_stream_get_parser(st);
 #if ((LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_6_V && LIBAVFORMAT_VERSION_MINOR >= LIBAVFORMAT_61_V) || LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_7_V)
@@ -3252,8 +3259,9 @@ GCC_DIAG_OFF(deprecated-declarations)
 			context->video_start_time, ticks, c ? c->ticks_per_frame : -1, st->time_base.num, st->time_base.den, c ? c->time_base.num : -1, c ? c->time_base.den : -1,
 			st->start_time, st->duration == AV_NOPTS_VALUE ? context->fc->duration / AV_TIME_BASE * 1000 : st->duration, st->nb_frames, av_q2d(st->time_base));
 	}
-#if ((LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_6_V && LIBAVFORMAT_VERSION_MINOR >= LIBAVFORMAT_61_V) || LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_7_V)
+#if ((LIBAVFORMAT_VERSION_MAJOR == LIBAVFORMAT_6_V && LIBAVFORMAT_VERSION_MINOR >= LIBAVFORMAT_61_V) || LIBAVFORMAT_VERSION_MAJOR >= LIBAVFORMAT_7_V)
 GCC_DIAG_ON(deprecated-declarations)
+#endif
 #endif
 
  again:
diff --git a/src/mod/applications/mod_av/mod_av.h b/src/mod/applications/mod_av/mod_av.h
index 8152a77..565537a 100644
--- a/src/mod/applications/mod_av/mod_av.h
+++ b/src/mod/applications/mod_av/mod_av.h
@@ -47,6 +47,7 @@
 #define LIBAVFORMAT_V 59 /* FFmpeg version >= 5.1 */
 #define LIBAVFORMAT_6_V 60 /* FFmpeg version >= 6.0 */
 #define LIBAVFORMAT_7_V 61 /* FFmpeg version >= 7.0 */
+#define LIBAVFORMAT_8_V 62 /* FFmpeg version >= 8.0 */
 #define LIBAVFORMAT_61_V 16 /* FFmpeg version >= 6.1 */
 #define LIBAVUTIL_V 57 /* FFmpeg version >= 5.1 */
 
