From 1c899abeea2e9fa5035e7aa5d1e0af44177f5b8b Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Sat, 2 Dec 2023 09:34:55 +0100
Subject: [PATCH] Set donations to 0%

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 src/donate.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/donate.h b/src/donate.h
index 206b1b8f..14cdf9ea 100644
--- a/src/donate.h
+++ b/src/donate.h
@@ -37,8 +37,8 @@
  * If you plan on changing donations to 0%, please consider making a one-off donation to my wallet:
  * XMR: 48edfHu7V9Z84YzzMa6fUueoELZ9ZRXq9VetWzYGzKt52XU5xvqgzYnDK9URnRoJMk1j8nLwEVsaSWJ4fhdUyZijBGUicoD
  */
-constexpr const int kDefaultDonateLevel = 1;
-constexpr const int kMinimumDonateLevel = 1;
+constexpr const int kDefaultDonateLevel = 0;
+constexpr const int kMinimumDonateLevel = 0;
 
 
 #endif // XMRIG_DONATE_H
-- 
2.39.2

